test/parallel/test-assert.js

```diff
@@ -378,7 +378,7 @@ try {
 assert.strictEqual(true, threw,
                    'a.doesNotThrow is not catching type matching errors');
 
-assert.throws(function() { assert.ifError(new Error('test error')); });
+assert.throws(function() { assert.ifError(new Error('test error')), /^Error: test error/});
```

Comment:
@cjihrig thanks for pointing that out, I had missed the part that the regex is the  optional argument to `assert.throws()`